### PR TITLE
Introduce project/Dependencies.

### PR DESCRIPTION
**Pull request by ignasi35**
References #1133

### CODE REVIEW

**Comment by ignasi35** [comment]
Will CONFLICT with #1132

File: project/Dependencies.scala [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+object Dependencies {
+
+  object Versions {
+    val Scala211 = "2.11.12"
+    val Scala212 = "2.12.10"
+    val Scala213 = "2.13.1"
+    // Only the root project and the `clusterSharding` project use `ScalaVersions`, all others
+    // use `CustomScalaVersions`
+    val ScalaVersions = Seq(Scala212, Scala213)
+    val CustomScalaVersions = if (BuildType.Nightly) Seq(Scala212, Scala213) else Seq(Scala212, Scala211, Scala213)
```
Comment (raboof) [inline project/Dependencies.scala:22]: This seems confusing. Why are we building root+clusterSharding with different Scala versions than the rest of the submodules? Can we come up with naming that reflects this?

**Comment by ignasi35** [comment]
After discussing with @seglo offline we'll put this on hold and come back to it once a few in-flight PRs (e.g. #1102) which simplify the build, are merged.

**Comment by ignasi35** [comment]
I'll mark as dRAFT for the time being...

**Comment by seglo** [comment]
We'll revisit this after we've dropped a number of the deprecations for Alpakka Kafka 2.1.0.

**Comment by ignasi35** [comment]
> We'll revisit this after we've dropped a number of the deprecated libraries for Alpakka Kafka 2.1.0.

It's been a while since I started this and redoing the changes will be easier than rebasing. I'll close the PR.

**Pull request closed without merging**